### PR TITLE
[Non-Inclusive Lanuage] Replace references to master branch with main

### PR DESCRIPTION
**Pull request by kaddy645**
Signed-off-by: kaddy645 <xdeskart@amazon.com>

### Description
 In this change, we are replacing references to master branch with main
 
### Issues Resolved
[https://github.com/opensearch-project/OpenSearch-Dashboards/issues/1691](https://github.com/opensearch-project/OpenSearch-Dashboards/issues/1691)
 
### Check List
- [ ] New functionality includes testing.
  - [x] All tests pass
    - [x] `yarn test:jest`
    - [x] `yarn test:jest_integration`
    - [x] `yarn test:ftr`
- [ ] New functionality has been documented.
- [X] Commits are signed per the DCO using --signoff

### CODE REVIEW

File: packages/osd-test/src/failed_tests_reporter/run_failed_tests_reporter_cli.ts [hunk 1]
```diff
@@ -65,9 +65,9 @@ export function runFailedTestsReporterCli() {
         }
 
         const isPr = !!process.env.ghprbPullId;
-        const isMasterOrVersion = branch === 'master' || branch.match(/^\d+\.(x|\d+)$/);
+        const isMasterOrVersion = branch === 'main' || branch.match(/^\d+\.(x|\d+)$/);
```
Comment (kavilla) [inline packages/osd-test/src/failed_tests_reporter/run_failed_tests_reporter_cli.ts:68]: is this the only place this constant is used? If so we can probably change the constant as well to `isMainOrVersion`

File: tasks/docker_docs/Dockerfile [hunk 2]
```diff
@@ -15,4 +15,4 @@ USER kibana
 RUN git clone --depth 1 https://github.com/elastic/docs.git /home/kibana/docs_builder
 
 WORKDIR /home/kibana/docs_builder
-CMD git pull origin master && ./build_docs.pl --doc /home/kibana/ascii_docs/index.asciidoc --out /home/kibana/html_docs --chunk=1
+CMD git pull origin main && ./build_docs.pl --doc /home/kibana/ascii_docs/index.asciidoc --out /home/kibana/html_docs --chunk=1
```
Comment (kavilla) [inline tasks/docker_docs/Dockerfile:18]: This file has a lot of legacy references. Would you be able to create a follow-up issue for us to clean this up/delete?

File: DEVELOPER_GUIDE.md [hunk 3]
```diff
@@ -341,7 +341,7 @@ assertion would now wrongly disable proper type checking for us.
 
 If you’re not using non-null assertions in your plugin or are starting a new plugin, consider enabling the
 [`@typescript-eslint/no-non-null-assertion`](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-non-null-assertion.md)
```
Comment (kavilla) [inline DEVELOPER_GUIDE.md:343]: I see `master` here and a link with `main` exists.

**Comment by kavilla** [comment]
Looking good so far! For the first pass just a few comments but I think this can be ready for review after addressing. If anything was missed we can just create subsequent issues.

For the git commit though can you update the commit description to include the details you include in the PR. For example, including the [issue](https://github.com/opensearch-project/OpenSearch-Dashboards/issues/1691).

Thanks @kaddy645!

File: packages/osd-spec-to-console/README.md [hunk 4]
```diff
@@ -13,7 +13,7 @@ git init
 git remote add origin https://github.com/opensearch-project/OpenSearch
 git config core.sparsecheckout true
 echo "rest-api-spec/src/main/resources/rest-api-spec/api/*\nx-pack/plugin/src/test/resources/rest-api-spec/api/*" > .git/info/sparse-checkout
```
[the remark this answers is not in the document]
Comment (kaddy645) [reply packages/osd-spec-to-console/README.md:15]: Created: https://github.com/opensearch-project/OpenSearch-Dashboards/issues/1783

File: tasks/docker_docs/Dockerfile (hunk 2 above)
Comment (kaddy645) [reply tasks/docker_docs/Dockerfile:18]: Created: https://github.com/opensearch-project/OpenSearch-Dashboards/issues/1784

**Review by joshuarrrr — APPROVED** [review]
Nice work, LGTM

**Review by kavilla — APPROVED** [review]

**Pull request merged**